hs_core/tasks.py

```diff
@@ -188,6 +190,8 @@ def manage_task_nightly():
                     res.doi = act_doi
                     res.save()
                     success = True
+                    # create bag and compute checksum for published resource to meet DataONE requirement
+                    create_bag_by_irods(res.short_id)
```

Comment:
Before you made this change, were bags created on the first download after publication?

Reply:
Yes. With this change, bags will be created with checksum computed before the first download after publication so that the published resources are ready to be consumed by DataONE.